### PR TITLE
how to do a Kubernetes rebase

### PR DESCRIPTION
**Pull request by danwinship**
It seemed like this belongs in `HACKING.md`, though the fact that we don't have anything else there yet makes it fit less well...

/assign @squeed

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+2. Update the remaining kubernetes and openshift dependencies in
+   `glide.yaml`. For dependencies under `github.com/openshift`, and
+   for kubernetes dependencies where we currently depend on a specific
+   commit SHA, see what version is now being used by the origin
+   rebase, and use that. (This is why you waited until after someone
+   started the origin rebase...)
+
+3. Try it out... run
+
+       glide update --strip-vendor
```
Comment (danwinship) [inline HACKING.md]: Ah, I was thinking there should be an automated way to do this and it turns out the answer is `make update-deps`
(@rcarrillocruz )

**Comment by danwinship** [comment]
Updated based on @rcarrillocruz 's experience:
- Clarified that the name of the kubernetes base branch should have the complete tag name in it. (eg, including the "`.0`" in this case)
- Clarified that there probably won't be any `<drop>` commits
- Updated with instructions on what to do if you can't push to `openshift/kubernetes`
- Clarified that we shouldn't depend on `master` when a module has no appropriate tag
- Fixed to say `make update-deps` rather than `glide update --strip-vendor` everywhere
- Added notes about possibly needing to explicitly pin new dependencies

**Comment by squeed** [comment]
This looks pretty good to me. @rcarrillocruz, want to give this another pass?

**Comment by danwinship** [comment]
/assign @rcarrillocruz 
It's already got /approved, so if you /lgtm it, it should merge

**Comment by squeed** [comment]
/lgtm
let's get this merged

**Comment by squeed** [comment]
/skip e2e-aws
/skip e2e-aws-upgrade
/skip e2e-gcp
the cloud providers don't need any more money than they already have

**Pull request merged**